### ISSUE TITLE
parse error with `TSImportType` with multiple import attributes

### ISSUE THREAD

**Issue by hi-ogawa** [oxc-project/oxc#9732]
- oxc playground https://playground.oxc.rs/#eNo9jz1PxDAMhv9K5AmkDoDEkhndCoL1Fjd1j6A0rhznrlXV/07Sqp389fp57QUcWKBpZFETSM1kjc4jcW/8UJtPV+iZr9CYxTy8/toSq8CaK/wljnVSBLXcdGY16zM0wGAXkBxrSHNUnMCqZGog+KhHnhyPdBbz0HI4KhWMqWcZwPYYEq0NjCiJpBIxBH58k2aJn1mT7+iSo1PPxW9fH4WK9k5fKBTT3i2Iar4hSn467ExFuVG5DCi9vby+lx984oBK3Qe5gIIVX0jHNY47utH2IEVsA/1wFkcDjqfb4KPv/eHnOKpwuJTT69KdpOVUvt+B6/oPFCSL+A==
- ts eslint https://ast-explorer.dev/#eNolzFEKwzAIgOGriE8bdBfIQfbkSyiWpaQxJLJ1hNx9uj6Jvx8OzBhwj+/Y15aq4oLVgn4rX+HBPafifbXOZ5WmkFnhDOBKNkiHxxvhJkK4wIBP0lew6SAA4d6l+MWAr38HE+bd3oq9HVTAOq9HfHLrybXDHJW7ElKZOH8pVzhO

```ts
export let x: typeof import("foo", { with: { type: "json", foo: "foo" } })
                                                              ^ Error: Expected `,` but found `:`
```

**Comment by Boshen** [comment]
This is a spec violation though 🤔 

TypeScript reports `Type import attributes should have exactly one key - 'resolution-mode' - with value 'import' or 'require'.`

**Comment by hi-ogawa** [comment]
Interesting, I didn't even know typescript had such feature built on top of import attributes 😅. It looks like that's something introduced during import assertion days https://github.com/microsoft/TypeScript/issues/49055.

I think typescript behavior might need to potentially change in the future depending on how they support import attributes ambient module https://github.com/microsoft/TypeScript/issues/46135.

**Comment by Boshen** [comment]
Our behaviour is correct according to the spec.
